origin.spec

```diff
@@ -0,0 +1,426 @@
+%{_bindir}/atomic-enterprise
+%{_bindir}/oadm
+%{_bindir}/kubectl
+%{_bindir}/kubernetes
+%{_bindir}/kubelet
+%{_bindir}/kube-proxy
+%{_bindir}/kube-apiserver
+%{_bindir}/kube-controller-manager
+%{_bindir}/kube-scheduler
+%{_sharedstatedir}/origin
+%{_sysconfdir}/bash_completion.d/*
+%dir %config(noreplace) %{_sysconfdir}/origin
+
+%pre
+# If /etc/openshift exists symlink it to /etc/origin
+if [ -d "%{_sysconfdir}/openshift" ]; then
+  ln -s %{_sysconfdir}/openshift %{_sysconfdir}/origin
+fi
+
+%files master
+%defattr(-,root,root,-)
+%{_unitdir}/%{name}-master.service
+%config(noreplace) %{_sysconfdir}/sysconfig/%{name}-master
+%config(noreplace) %{_sysconfdir}/origin/master
+%ghost %config(noreplace) %{_sysconfdir}/origin/admin.crt
+%ghost %config(noreplace) %{_sysconfdir}/origin/admin.key
+%ghost %config(noreplace) %{_sysconfdir}/origin/admin.kubeconfig
+%ghost %config(noreplace) %{_sysconfdir}/origin/ca.crt
+%ghost %config(noreplace) %{_sysconfdir}/origin/ca.key
+%ghost %config(noreplace) %{_sysconfdir}/origin/ca.serial.txt
+%ghost %config(noreplace) %{_sysconfdir}/origin/etcd.server.crt
+%ghost %config(noreplace) %{_sysconfdir}/origin/etcd.server.key
+%ghost %config(noreplace) %{_sysconfdir}/origin/master-config.yaml
+%ghost %config(noreplace) %{_sysconfdir}/origin/master.etcd-client.crt
+%ghost %config(noreplace) %{_sysconfdir}/origin/master.etcd-client.key
+%ghost %config(noreplace) %{_sysconfdir}/origin/master.kubelet-client.crt
+%ghost %config(noreplace) %{_sysconfdir}/origin/master.kubelet-client.key
+%ghost %config(noreplace) %{_sysconfdir}/origin/master.server.crt
+%ghost %config(noreplace) %{_sysconfdir}/origin/master.server.key
+%ghost %config(noreplace) %{_sysconfdir}/origin/openshift-master.crt
+%ghost %config(noreplace) %{_sysconfdir}/origin/openshift-master.key
+%ghost %config(noreplace) %{_sysconfdir}/origin/openshift-master.kubeconfig
+%ghost %config(noreplace) %{_sysconfdir}/origin/openshift-registry.crt
+%ghost %config(noreplace) %{_sysconfdir}/origin/openshift-registry.key
+%ghost %config(noreplace) %{_sysconfdir}/origin/openshift-registry.kubeconfig
+%ghost %config(noreplace) %{_sysconfdir}/origin/openshift-router.crt
+%ghost %config(noreplace) %{_sysconfdir}/origin/openshift-router.key
+%ghost %config(noreplace) %{_sysconfdir}/origin/openshift-router.kubeconfig
+%ghost %config(noreplace) %{_sysconfdir}/origin/policy.json
+%ghost %config(noreplace) %{_sysconfdir}/origin/serviceaccounts.private.key
+%ghost %config(noreplace) %{_sysconfdir}/origin/serviceaccounts.public.key
+
+%post master
+# TODO: remove this conditional at a later date
+%systemd_post %{basename:%{name}-master.service}
+# Create master configs if they do not exist
+if [ ! -e %{_sysconfdir}/origin/master/master-config.yaml ]; then
+%if "%{dist}" == ".el7aos"
+  %{_bindir}/atomic-enterprise start master --write-config=%{_sysconfdir}/origin/master
+%else
+  %{_bindir}/openshift start master --write-config=%{_sysconfdir}/origin/master
+%endif
+fi
+
```

Comment:
@sdodson I am trying to understand this: how would it work? How is it possible to choose the binary just based on .el7aos because above in the same spec file you do:

%if "%{dist}" == ".el7aos"
%global package_name atomic-openshift
%global product_name Atomic OpenShift
%else
%global package_name origin
%global product_name Origin
%endif

That means for el7aos, it could be both %{_bindir}/atomic-enterprise or %{_bindir}/openshift, so you need some other conditions to choose among binaries. 

My guess is that you should choose the package name based on .el7ose and .el7aos so something like:

%if "%{dist}" == ".el7aos"
%global package_name atomic-openshift
%global product_name Atomic
%if else "%{dist}" == ".el7ose"
%global package_name atomic-openshift
%global product_name OpenShift
%else
%global package_name origin
%global product_name Origin
%endif